### PR TITLE
Added final declarations to constants (#5275)

### PR DESCRIPTION
**Pull request by WisdomPill**
## What do these changes do?

Adds final declarations to constants

## Are there changes in behavior for the user?
Let the user know that these constants should be considered as immutable

## Related issue number
#5275

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] Add a new news fragment into the `CHANGES` folder

### CODE REVIEW

**Comment by WisdomPill** [comment]
@asvetlov `Final` has been introduced since python 3.8, what to do then?

File: aiohttp/client.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
     Awaitable,
     Callable,
     Coroutine,
+    Final,
```
Comment (asvetlov) [inline aiohttp/client.py]: Please import `Final` from `typing_extensions` for backward compatibility with 3.6 and 3.7

File: aiohttp/client_ws.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 import asyncio
 import dataclasses
-from typing import Any, Optional
+from typing import Any, Final, Optional
```
Comment (asvetlov) [inline aiohttp/client_ws.py]: The same

File: aiohttp/hdrs.py [hunk 3]
```diff
@@ -2,21 +2,22 @@
 
 # After changing the file content call ./tools/gen.py
 # to regenerate the headers parser
+from typing import Final, Set
```
Comment (asvetlov) [inline aiohttp/hdrs.py]: typing_extensions

File: aiohttp/http_parser.py [hunk 4]
```diff
@@ -6,7 +6,19 @@
 import zlib
 from contextlib import suppress
 from enum import IntEnum
-from typing import Any, Generic, List, Optional, Tuple, Type, TypeVar, Union
+from typing import (
+    Any,
+    Final,
```
Comment (asvetlov) [inline aiohttp/http_parser.py]: typing_extensions

File: aiohttp/http_websocket.py [hunk 5]
```diff
@@ -9,7 +9,7 @@
 import zlib
 from enum import IntEnum
 from struct import Struct
-from typing import Any, Callable, List, Optional, Tuple, Union
+from typing import Any, Callable, Final, List, Optional, Pattern, Set, Tuple, Union
```
Comment (asvetlov) [inline aiohttp/http_websocket.py]: typing_extensions

File: aiohttp/payload.py [hunk 6]
```diff
@@ -52,7 +53,7 @@
     "AsyncIterablePayload",
 )
 
-TOO_LARGE_BYTES_BODY = 2 ** 20  # 1 MB
+TOO_LARGE_BYTES_BODY: Final[int] = 2 ** 20  # 1 MB
```
Comment (asvetlov) [inline aiohttp/payload.py:56]: typing_extensions

File: aiohttp/web_fileresponse.py [hunk 7]
```diff
@@ -9,6 +9,7 @@
     Any,
     Awaitable,
     Callable,
+    Final,
```
Comment (asvetlov) [inline aiohttp/web_fileresponse.py]: typing_extensions

File: aiohttp/web_request.py [hunk 8]
```diff
@@ -14,10 +14,12 @@
     TYPE_CHECKING,
     Any,
     Dict,
+    Final,
```
Comment (asvetlov) [inline aiohttp/web_request.py]: typing_extensions

File: aiohttp/web_urldispatcher.py [hunk 9]
```diff
@@ -14,6 +14,7 @@
     Callable,
     Container,
     Dict,
+    Final,
```
Comment (asvetlov) [inline aiohttp/web_urldispatcher.py]: typing_extensions

File: aiohttp/web_ws.py [hunk 10]
```diff
@@ -4,7 +4,7 @@
 import dataclasses
 import hashlib
 import json
-from typing import Any, Iterable, Optional, Tuple
+from typing import Any, Final, Iterable, Optional, Tuple
```
Comment (asvetlov) [inline aiohttp/web_ws.py]: typing_extensions

**Review by asvetlov — CHANGES_REQUESTED** [review]

**Comment by WisdomPill** [comment]
@asvetlov can you guide on the cython error that comes up now?

**Comment by WisdomPill** [comment]
@asvetlov friendly reminder, I get an error in cythonize, I do not know how to solve it

**Review by asvetlov — APPROVED** [review]

**Comment by asvetlov** [comment]
To make CI passed please add the following line to `requirements/cython.txt`: 
```
typing_extensions==3.7.4.3  # required for parsing aiohttp/hdrs.py by tools/gen.py
```

**Comment by asvetlov** [comment]
Thanks!

**Pull request merged**